okhttp-clients/src/test/java/com/palantir/remoting3/okhttp/UrlSelectorTest.java

```diff
@@ -123,6 +126,17 @@ public void testRedirectToNext() throws Exception {
         assertThat(selector.redirectToNext(current)).contains(HttpUrl.parse("http://foo/a/b/foo"));
     }
 
+    @Test
+    public void testWorksWithWebSockets() throws Exception {
+        Request wsRequest = new Request.Builder()
+                .url("wss://foo/a")
+                .build();
+        UrlSelectorImpl selector = UrlSelectorImpl.create(ImmutableList.of("wss://foo/", "wss://bar/"));
+
+        // Silently replace web socket URLs with HTTP URLs. See https://github.com/square/okhttp/issues/1652.
+        assertThat(selector.redirectToNext(wsRequest.url())).isEqualTo(Optional.of(parse("https://bar/a")));
```

Comment:
Where is the "ws" used at all? I find it a bit funny to return "http://foo" as the "next" url for an initial url "ws://foo".

Reply:
Technically you're not doing this since the URL on the Request class is `http://foo` since OkHTTP converts transparently under the hood.

Reply:
See the code I linked above